### PR TITLE
Update GitHub Actions

### PR DESCRIPTION
**Pull request by davidkircos**
- [x] Moves all GH actions to a single easy to maintain file
- [x] Speed up CI!
- [x] Verify new tests do everything the previous ones do
- [x] report coverage
- [x] backup in case `self-hosted` is down
- [x] Verify optimization happens when it should

With our self-hosted runners all online tests run fast:
CleanShot 2023-12-03 at 21 43 34@2x

All our self-hosted runners are tagged `ubuntu-latest-8-cores` if our self-hosted runners are not available or if more jobs are run than the eight available. GitHub will spin up hosted runners with the same tag `ubuntu-latest-8-cores`. 

With our self-hosted runners all offline. Tests still pass but take longer:
CleanShot 2023-12-03 at 17 17 20@2x

This is the current speed of actions on `main`
CleanShot 2023-12-04 at 10 07 06@2x

### CODE REVIEW

File: vercel.json [hunk 1]
```diff
@@ -1,4 +0,0 @@
-{
-  "buildCommand": "cd quadratic-client && npm run build:ci",
-  "installCommand": "cd quadratic-client && npm install"
-}
```
Comment (davidkircos) [inline vercel.json:4]: Pretty sure this is unused as it is defined in the vercel ui

File: quadratic-core/Cargo.toml [hunk 2]
```diff
@@ -83,6 +83,8 @@ harness = false
 # Tell `rustc` to optimize for small code size.
 opt-level = "s"
 
+[package.metadata.wasm-pack.profile.release]
+wasm-opt = ['-Os', '-g']  # TODO: -g seems to fix the name mangling problem
```
Comment (davidkircos) [inline quadratic-core/Cargo.toml:87]: see 🧵 https://quadratichq.slack.com/archives/C03VC0Q1MPA/p1701655393730929

File: quadratic-api/package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "lint:prettier:write": "prettier --write \"src/**/*.ts\"",
     "prisma:migrate": "npx prisma migrate dev",
     "prisma:migrate:create": "npx prisma migrate dev --create-only",
-    "docker:up": "docker-compose up -d && sleep 1",
+    "docker:up": "docker-compose up -d && sleep 3",
```
Comment (ddimaria) [inline quadratic-api/package.json:16]: Curious, what's the need for this `sleep`?

File: amplify.yml [hunk 4]
```diff
@@ -11,7 +11,8 @@ frontend:
 
     build:
       commands:
-        - npm run build:wasm
```
Comment (ddimaria) [inline amplify.yml:14]: Is this already built in GHA?

**Review by ddimaria — APPROVED** [review]
Looks great!  I just had a couple of questions, but nothing blocking.

File: quadratic-api/package.json (hunk 3 above)
Comment (davidkircos) [reply quadratic-api/package.json:16]: It takes a sec for Postgres to be available. Commands that actually wait for Postgres to be up are really complicated. This works 🤷‍♂️

File: amplify.yml (hunk 4 above)
Comment (davidkircos) [reply amplify.yml:14]: Amplify is a separate pipeline

**Pull request merged**